features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -171,7 +171,9 @@ void AT_CellularNetwork::read_reg_params_and_compare(RegistrationType type)
                     reg_params._status == RegisteredRoaming)) {
                 if (previous_registration_status == RegisteredHomeNetwork ||
                         previous_registration_status == RegisteredRoaming) {
-                    call_network_cb(NSAPI_STATUS_DISCONNECTED);
+                    if (type != C_REG) {// we are interested only if we drop from packet network
+                        _connection_status_cb(NSAPI_EVENT_CONNECTION_STATUS_CHANGE, NSAPI_STATUS_DISCONNECTED);
```

Comment:
Is this an API break? Functionality changes.

Reply:
This change is coming from PR #9937.

Reply:
As we have a default value, the user can override that with NULL to "disable" it. What would go wrong if you call set_timeout with NULL and in function body does nothing in that case? User can call that with NULL as well, so we should have NULL check already in place, or?